android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java

```diff
@@ -117,6 +120,47 @@ public void onDismiss() {
         setUpViews();
     }
 
+    public void showPopupMenu(){
+       PopupMenu menu = new PopupMenu(mOptionsImage.getContext(),   (View) mOptionsImage);
+       menu.getMenuInflater().inflate(R.menu.menu_dapps_panel, menu.getMenu());
+       menu.setOnMenuItemClickListener(item -> {
+            if (item.getItemId() == R.id.action_lock_wallet) {
+                mBraveWalletPanelServices.getKeyringService().lock();
+                dismiss();
+            }
+            if (item.getItemId() == R.id.action_connected_sites) {
+                //TODO needs to be implemented in another ticket
+                dismiss();
+            }
+            if (item.getItemId() == R.id.action_settings) {
+                BraveActivity activity = BraveActivity.getBraveActivity();
+                if (activity != null) {
+                    activity.openBraveSettings();
+                    dismiss();
+                }
+            }
+            if (item.getItemId() == R.id.action_view_on_block_explorer) {
+                BraveActivity activity = BraveActivity.getBraveActivity();
+                if (activity != null) {
+                    mBraveWalletPanelServices.getKeyringService().getKeyringInfo(
+                    BraveWalletConstants.DEFAULT_KEYRING_ID, keyringInfo -> {
+                        if (keyringInfo != null) {
+                            mAccountInfos = keyringInfo.accountInfos;
+                            if (mAccountInfos.length > 0)
+                                activity.viewOnBlockExplorer(mAccountInfos[0].address);
+                            dismiss();
+                        }
+                    });
```

Comment:
Shouldn't `getKeyringService().getSelectedAccount` be used here to get the address? as first account can be any account?

Reply:
same, not resolved, changing back to Unresolve.